lntest/itest/lnd_forward_interceptor_test.go

```diff
@@ -204,266 +200,161 @@ func testForwardInterceptorDedupHtlc(net *lntest.NetworkHarness, t *harnessTest)
+
+		payment := ht.AssertPaymentStatus(
+			ctx.alice, preimage, lnrpc.Payment_IN_FLIGHT,
+		)
+		expectedAmt := testCase.invoice.ValueMsat
+		require.Equal(ht, expectedAmt, payment.ValueMsat,
+			"incorrect in flight amount")
 	}
 
-	// Disconnect interceptor should cause resume held packets.
-	// After that we wait for all go routines to finish, including the one
-	// that tests the payment final status for the held payment.
+	// Cancel the context, which will disconnect the above interceptor.
 	cancelInterceptor()
-	wg.Wait()
+
+	// Disconnect interceptor should cause resume held packets. After that
+	// we wait for all go routines to finish, including the one that tests
+	// the payment final status for the held payment.
+	select {
+	case <-done:
+	case <-time.After(defaultTimeout):
+		require.Fail(ht, "timeout waiting for sending payment")
+	}
 
 	// Verify that we don't get notified about already completed HTLCs
 	// We do that by restarting alice, the sender the HTLCs. Under
 	// https://github.com/lightningnetwork/lnd/issues/5115
-	// this should cause all HTLCs settled or failed by the interceptor to renotify.
-	restartAlice, err := net.SuspendNode(alice)
-	require.NoError(t.t, err, "failed to suspend alice")
+	// this should cause all HTLCs settled or failed by the interceptor to
+	// renotify.
+	restartAlice := ht.SuspendNode(ctx.alice)
+	require.NoError(ht, restartAlice(), "failed to restart alice")
 
-	ctxt, cancelInterceptor = context.WithTimeout(ctxb, defaultTimeout)
-	defer cancelInterceptor()
-	interceptor, err = testContext.bob.RouterClient.HtlcInterceptor(ctxt)
-	require.NoError(t.t, err, "failed to create HtlcInterceptor")
+	// Make sure the channel is active from Bob's PoV.
+	ht.AssertChannelExists(ctx.bob, chanPointAliceBob)
 
-	err = restartAlice()
-	require.NoError(t.t, err, "failed to restart alice")
+	// Create a new interceptor as the old one has quit.
+	interceptor, cancelInterceptor = ht.HtlcInterceptor(ctx.bob)
 
+	done = make(chan struct{})
 	go func() {
-		request, err := interceptor.Recv()
-		if err != nil {
-			// If it is  just the error result of the context cancellation
-			// the we exit silently.
-			status, ok := status.FromError(err)
-			if ok && status.Code() == codes.Canceled {
-				return
-			}
-			// Otherwise it an unexpected error, we fail the test.
-			require.NoError(
-				t.t, err, "unexpected error in interceptor.Recv()",
-			)
+		_, err := interceptor.Recv()
+
+		require.Error(ht, err, "expected an error from interceptor")
+
+		status, ok := status.FromError(err)
+		switch {
+		// If it is just the error result of the context cancellation
+		// the we exit silently.
+		case ok && status.Code() == codes.Canceled:
+			fallthrough
+
+		// When the test ends, during the node's shutdown it will close
+		// the connection.
+		case strings.Contains(err.Error(), "closed network connection"):
+			fallthrough
+
+		case strings.Contains(err.Error(), "EOF"):
+			close(done)
```

Comment:
Use `defer` in combination with `require` (see other comments).

Reply:
Fixed!